packages/@aws-cdk/aws-iotevents-actions/lib/set-timer-action.ts

```diff
@@ -0,0 +1,77 @@
+import * as iotevents from '@aws-cdk/aws-iotevents';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+/**
+ * Configuration properties of an action to set a timer.
+ */
+export interface SetTimerActionProps {
+  /**
+   * The duration of the timer, in seconds. One of `duration` or `durationExpression` is required.
+   *
+   * The range of the duration is 60-31622400 seconds.
+   * The evaluated result of the duration expression is rounded down to the nearest whole number.
+   * For example, if you set the timer to 60.99 seconds, the evaluated result of the duration expression is 60 seconds.
+   *
+   * @default - none, required if no `durationExpression` is defined.
+   */
+  readonly duration?: cdk.Duration;
```

Comment:
I'm not a fan of mutually exclusive props. I think this should go in the direction of one required prop that gets translated properly into `Duration` or `Expression`. Take a look at the model of this we use for [Schedule](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-events/lib/schedule.ts) in `aws-events`.

Reply:
Cool! 🤩 I will implement as it!